tests/usage/test_usage_lib.py

```
@@ -0,0 +1,214 @@
+from tensorrt_llm.usage.usage_lib import UsageMessage, collect_system_info, detect_cloud_provider
+
+
+class TestOptOut:
+    """Test opt-out mechanisms."""
+
+    def test_opt_out_via_env_var(self, monkeypatch):
+        """Test opt-out via TRTLLM_NO_USAGE_STATS environment variable."""
+        monkeypatch.setenv("TRTLLM_NO_USAGE_STATS", "1")
+        assert not is_usage_stats_enabled()
+
+    def test_opt_out_via_file(self, tmp_path, monkeypatch):
```

Ruff (0.14.14)

[warning] Unused method argument: `monkeypatch`

(ARG002)